### PR TITLE
Feat: Component for setting business hour availability

### PR DESCRIPTION
**Pull request by nithindavid**
Fixes part of #234

### CODE REVIEW

File: app/javascript/dashboard/i18n/locale/en/inboxMgmt.json [hunk 1]
```diff
@@ -226,7 +225,8 @@
       "SETTINGS": "Settings",
       "COLLABORATORS": "Collaborators",
       "CONFIGURATION": "Configuration",
-      "PRE_CHAT_FORM": "Pre Chat Form"
+      "PRE_CHAT_FORM": "Pre Chat Form",
+      "BUSINESSHOURS": "Business Hours"
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/inboxMgmt.json:229]: BUSINESS_HOURS?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    </div>
+    <div v-if="isDayEnabled" class="hours-select-wrap">
+      <div class="hours-range">
+        <multiselect
+          v-model="fromTime"
+          :options="timeSlots"
+          deselect-label=""
+          select-label=""
+          selected-label=""
+          placeholder="Choose"
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:23]: i18n.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        <div class="separator-icon">
+          <i class="ion-minus-round" />
+        </div>
+        <multiselect
+          v-model="toTime"
+          :options="timeSlots"
+          deselect-label=""
+          select-label=""
+          selected-label=""
+          placeholder="Choose"
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:35]: i18n.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+.label {
+  font-size: var(--font-size-mini);
+  color: var(--w-700);
+  background: var(--w-50);
+}
+
+.date-error {
+  padding: var(--space-small) 0;
+  /* position: absolute;
+  bottom: var(--space-minus-medium); */
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:218]: 👃

File: app/javascript/dashboard/routes/dashboard/settings/inbox/helpers/businessHour.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const generateTimeSlots = (step = 15) => {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/helpers/businessHour.js:1]: Please add test cases

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    </div>
+  </div>
+</template>
+
+<script>
+import parse from 'date-fns/parse';
+import differenceInMinutes from 'date-fns/differenceInMinutes';
+import { generateTimeSlots } from '../helpers/businessHour';
+
+const timeSlots = generateTimeSlots(30);
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:63]: let us put 15 minute interval.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    isDayEnabled: {
+      get() {
+        return this.timeSlot.from && this.timeSlot.to;
+      },
+      set(value) {
+        const newSlot = value
+          ? {
+              ...this.timeSlot,
+              from: timeSlots[0],
+              to: timeSlots[16],
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:94]: I don't understand this logic. Why is it `timeSlots[16]`?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+      },
+    },
+    fromDate() {
+      return parse(this.fromTime, 'hh:mm a', new Date());
+    },
+    toDate() {
+      return parse(this.toTime, 'hh:mm a', new Date());
+    },
+    totalHours() {
+      return differenceInMinutes(this.toDate, this.fromDate) / 60;
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:133]: Should we use toFixed here?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue (hunk 7 above)
Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:94]: Setting 8 hour gap as default.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/helpers/businessHour.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  const date = new Date(1970, 0, 1);
+  const slots = [];
+  while (date.getDate() === 1) {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/helpers/businessHour.js:4]: I don't understand the logic here. Why do we check for the date?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue (hunk 8 above)
Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:133]: This is mainly to show total hours,  to show 7.5 hours

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+    timeSlot: {
+      type: Object,
+      default: () => ({
+        from: '',
+        to: '',
+      }),
+    },
+  },
+  computed: {
+    timeSlots() {
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:82]: This is an unnecessary method.

Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/components/BusinessDay.vue:82]: The constant `timeSlots` needs to be accessed in template

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**